src/plone/restapi/deserializer/dxcontent.py

```diff
@@ -78,7 +78,14 @@ def __call__(self, validate_all=False, data=None):  # noqa: ignore=C901
                             'message': e.doc(), 'field': name, 'error': e})
                     else:
                         field_data[name] = value
-                        if value != dm.get():
+                        try:
+                            if value != dm.get():
+                                dm.set(value)
+                                modified = True
+                        except TypeError:
```

Comment:
I am sorry, but I did not like the usage of an Error Handler to apply a value

I would suggest something like:

    dm_value = dm.get()
    if value.tzinfo is None:
        value = value.replace(tzinfo=pytz.UTC)
    else:
         value = value.asdatetime(pytz.UTC)
     if dm_value.tzinfo is None:
        dm_value = dm_value.replace(tzinfo=pytz.UTC)
    else:
         dm_value = dm_value.asdatetime(pytz.UTC)

    if value != dm_value:
        dm.set(value)
        modified = True
 

instead of the try except clause.

Reply:
Ok, but you have to detect first if dm.get() returns a datetime right? if not, value.tzinfo will fail. I wanted to prevent to infer more conditional clauses here, which in turn, increases the complexity ratio of the method (and then plone.recipe.codeanalysis complains about it).

Reply:
yes you are right, there needs to be an ``if isinstance(value, datetime) and isinstance(dm_value, datetime):``  that wraps both if else clause.

Reply:
Well the complexity of an aditional if level is better than an try except. It could also be an idea to split this test aus into a separate function.

Reply:
@sneridagh @loechel For me the main objective of this code snippet is readability. I don't mind a try/except if it makes the code more readable. Though, without a few more comments I don't really understand what goes on here anyways. :)